tools/ddoc_xml_extractor.d

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env rdmd
+/*
+ * Extracts the newest articles from dlang.org's XML site and
+ * converts them to Ddoc macros.
+ *
+ * Copyright (C) 2018 by D Language Foundation
+ *
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+*/
+// Written in the D programming language.
+
+
+import std.format;
+
+struct Article
+{
+    string title;
+    string link;
+    string date;
+    string author;
+}
+
+auto toDdoc(const ref Article article, string suffix)
+{
+    string text;
+with(article)
+    text =
+`DBLOG_LATEST_TITLE%5$s=%1$s
+DBLOG_LATEST_LINK%5$s=%2$s
+DBLOG_LATEST_DATE%5$s=%3$s
+DBLOG_LATEST_AUTHOR%5$s=%4$s`.format(title, link, date, author, suffix);
```

Comment:
Any of these need HTML/XML escaping?

Reply:
It looks to me like they are already escaped:

```
<title>DConf 2018 Programme &amp; Open Registration</title>
<link>https://dlang.org/blog/2018/03/19/dconf-2018-programme-open-registration/</link>
<comments>https://dlang.org/blog/2018/03/19/dconf-2018-programme-open-registration/#respond</comments>
<pubDate>Mon, 19 Mar 2018 13:27:32 +0000</pubDate>
<dc:creator><![CDATA[Michael Parker]]></dc:creator>
```

Reply:
> `<title>DConf 2018 Programme &amp; Open Registration</title>`

Won't std.xml parse that `&amp;` into an `&`, meaning we now have to escape it back into an `&amp;` ?

Reply:
Yes you are right though I don't think we have HTML escaping out-of the box in Phobos, do we?
One could also make the point that dlang.org uses `<meta charset="utf-8">` and HTML escaping for non-ASCII characters isn't required.

Reply:
`std.xml.encode` ?

Reply:
>  One could also make the point that dlang.org uses <meta charset="utf-8"> and HTML escaping for non-ASCII characters isn't required.

Right, not unicode, just `&amp;` and `&gt;`

Reply:
Ah nice! Didn't know about that one. Done!